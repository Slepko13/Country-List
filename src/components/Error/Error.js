import React from "react";
import './Error.scss';
import PropTypes from "prop-types";


const Error = ({message}) => {
    return (
        <div className="Error">
            Sorry,  you got an error: {message || "some error"}
        </div>
    )
}
export default Error;
Error.propTypes = {
    message: PropTypes.string
}