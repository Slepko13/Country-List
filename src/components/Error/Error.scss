.Error {
  @media screen and (max-width: 991px){
    display: none;
  }
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  font-size: 36px;
  color:red;
}