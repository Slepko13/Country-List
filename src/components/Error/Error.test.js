import React from "react";
import Error from "./Error";

const setUp = (props) => shallow(<Error {...props}/>)

describe("my Error component", () => {
    let component;
    beforeEach(()=>{
        component = setUp();
    });

    it("should render Error Component ", () => {
        const wrapper = component.find(".Error");
        expect(wrapper.length).toBe(1);
        expect(component).toMatchSnapshot();
    });

    it("should render Error Component with props(snapshot)", () => {
        let component = shallow(<Error message="Server error"/>);
        expect(component).toMatchSnapshot();
    });

    it("should render Error Component without props(snapshot)", () => {
        let component = shallow(<Error />);
        expect(component).toMatchSnapshot();
    });
});

