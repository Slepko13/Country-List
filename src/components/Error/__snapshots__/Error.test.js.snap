// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my Error component should render Error Component  1`] = `
<div
  className="Error"
>
  Sorry,  you got an error: 
  some error
</div>
`;

exports[`my Error component should render Error Component with props(snapshot) 1`] = `
<div
  className="Error"
>
  Sorry,  you got an error: 
  Server error
</div>
`;

exports[`my Error component should render Error Component without props(snapshot) 1`] = `
<div
  className="Error"
>
  Sorry,  you got an error: 
  some error
</div>
`;
